flytekit/common/tasks/task.py

```diff
@@ -268,6 +273,58 @@ def _python_std_input_map_to_literal_map(self, inputs):
             for k, v in _six.iteritems(self.interface.inputs)
         })
 
+    def _produce_deterministic_version(self, version=None):
+        """
+        :param Text version:
+        :return Text:
+        """
+
+        if self.container is not None:
```

Comment:
This won't work for Ketan's [raw container tasks](https://github.com/lyft/flytekit/pull/107/files#diff-0b4aabc3c6488fa90c0e7f3afca65ae7R214) btw.  How about a whitelist of allowed task types (strings) instead?

Reply:
@wild-endeavor raw container tasks are not task type. You just have to check a different, thing, as this proposal
```suggestion
        if self.container is not None and self.container. data_config is None:
```

Reply:
Is it true that client side task versions do not work with container tasks? why not? I thought that's exactly what we do today... (we grab the Image version and make it the task version)...

Reply:
thanks for the catch/suggestion! done